marshal.go

```diff
@@ -82,6 +88,29 @@ func (d *Decoder) unmarshal() interface{} {
 	}
 }
 
+func (d *Decoder) parseFloat() float64 {
+	var result float64
+	var buff bytes.Buffer
+	b, errReadB := d.r.ReadByte()
+	if errReadB != nil {
+		result = 0.00
+	}
+	length := int(b) - 5
+
+	buf := make([]byte, length)
+	_, errReadF := io.ReadFull(d.r, buf)
+	if errReadF != nil {
+		result = 0.00
+	}
+	d.r = bufio.NewReader(&buff)
+	val := string(buf[:])
+	result, err := strconv.ParseFloat(val, 64)
```

Comment:
Does strconv.ParseFloat handle \bnan, \binf and \t-inf? Looks like we write those strings in those unusual cases in encFloat.

Reply:
It should handle all of those: https://pkg.go.dev/strconv#ParseFloat